crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -436,47 +592,22 @@ impl<'db> TypeInferenceBuilder<'db> {
         Type::Union(union)
     }
 
+    #[allow(clippy::unused_self)]
```

Comment:
Is this allow still necessary. It seems `self` is used in the method body

Reply:
Would love to have the nightly feature `expect` instead of `allow` to force cleaning these up.